asttokens/mark_tokens.py

```diff
@@ -237,12 +234,9 @@ def visit_subscript(self, node, first_token, last_token):
 
   def handle_bare_tuple(self, node, first_token, last_token):
     # A bare tuple doesn't include parens; if there is a trailing comma, make it part of the tuple.
-    try:
-      maybe_comma = self._code.next_token(last_token)
-      if util.match_token(maybe_comma, token.OP, ','):
-        last_token = maybe_comma
-    except IndexError:
```

Comment:
Same here.

Reply:
`next_token` can't raise an error here because there's always an endmarker in tokens, which I've just added a check for. If `last_token` is already the endmarker then that's a problem which should be fixed. I think our extensive tests, particularly for tuples, are sufficient.